frontend/src/features/modules/ModulesTree.tsx

```diff
@@ -1,60 +1,152 @@
+  if (!decl.value || !decl.value.case || !decl.value.value) {
+    return []
+  }
+  const Icon = icons[decl.value.case] || CodeBracketSquareIcon
+  return (
+    <li className='my-1'>
+      <DisclosureButton
+        as='a'
+        href={href}
+        className={classNames('hover:bg-gray-100 hover:dark:bg-gray-700', 'group flex items-center gap-x-2 rounded-md pl-4 pr-2 text-sm font-light leading-6')}
+      >
+        <Icon aria-hidden='true' className='size-4 shrink-0' />
+        {decl.value.value.name}
+        {(decl.value.value as WithExport).export === true ? <ExportBadge /> : []}
+      </DisclosureButton>
+    </li>
+  )
+}
+
+const ModuleSection = ({
+  module,
+  isExpanded,
+  path,
+  params,
+  toggleExpansion,
+}: { module: ModuleTreeItem; isExpanded: boolean; path: string; params: string; toggleExpansion: (m: string) => void }) => {
+  return (
+    <li key={module.name} id={`module-tree-module-${module.name}`} className='my-2'>
+      {module.decls.length === 0 ? (
+        <a
+          href={`${path}?${params}`}
+          className={classNames('hover:bg-gray-100 hover:dark:bg-gray-700', 'group flex gap-x-3 rounded-md px-2 text-sm font-medium leading-6')}
+        >
+          <BookOpenIcon aria-hidden='true' className='size-3 shrink-0' />
+          {module.name}
+        </a>
+      ) : (
+        <Disclosure as='div' defaultOpen={isExpanded}>
+          <DisclosureButton
+            className={classNames(
+              'hover:bg-gray-100 hover:dark:bg-gray-700',
+              'group flex w-full modules-center gap-x-2 space-y-1 rounded-md px-2 text-left text-sm font-medium leading-6',
+            )}
+            onClick={() => toggleExpansion(module.name)}
+          >
+            <BookOpenIcon aria-hidden='true' className='size-4 my-1 shrink-0 ' />
+            {module.name}
+            <a href={`${path}?${params}`}>
+              <ArrowRightCircleIcon
+                aria-hidden='true'
+                className='size-4 shrink-0 rounded-md hover:bg-gray-200 dark:hover:bg-gray-600'
+                onClick={(e) => {
+                  e.stopPropagation()
+                }}
+              />
+            </a>
+            <ChevronRightIcon aria-hidden='true' className='ml-auto h-4 w-4 shrink-0 group-data-[open]:rotate-90 group-data-[open]:text-gray-500' />
+          </DisclosureButton>
+          <DisclosurePanel as='ul' className='px-2'>
+            {module.decls.map((d, i) => (
+              <DeclNode key={i} decl={d} href={`${path}/${d.value.case}/${d.value.value?.name}?${params}`} />
+            ))}
+          </DisclosurePanel>
+        </Disclosure>
+      )}
+    </li>
+  )
+}
+
 export const ModulesTree = ({ modules }: { modules: ModuleTreeItem[] }) => {
+  const [searchParams, setSearchParams] = useSearchParams()
+  modules.sort((m1, m2) => Number(m1.isBuiltin) - Number(m2.isBuiltin))
+
+  const expandedModules = (searchParams.get('tree_m') || '').split(',')
```

Comment:
I think this behavior is a bit different from what I had in mind. I don't think we need to be able to share the tree with a set of expanded modules. Instead, I think we should share the tree with the `selected` node. That way when I share a `url` with you, it will load the tree with that module selected (and show it selected) and show the detail page associated with that node.

Reply:
Ahhhh ok that makes more sense. I was thinking this is pretty nonstandard. That makes this a lot simpler.

Reply:
replaced w localStorage